review/get_review_ids.py

```diff
@@ -0,0 +1,212 @@
+        parent = course_blocks.get_parents(record.module_state_key)[0]
+        vertical_id = str(parent).split("@")[-1]
+
+        vertical_data.add(vertical_id)
+    if len(vertical_data) < 1:
+        return []
+
+    vertical_to_show = random.sample(vertical_data, 1)[0]
+    review_course_id = REVIEW_COURSE_MAPPING[str(current_course)]
+    return (TEMPLATE_URL.format(course_id=review_course_id,
+                    type='vertical', xblock_id=vertical_to_show))
+
+def enroll_user(user, current_course):
+    '''
+    If the user is not enrolled in the review version of the course,
+    they are unable to see any of the problems. This ensures they
+    are enrolled so they can see review problems.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+    '''
+    enrollment_course_id = ENROLLMENT_COURSE_MAPPING[str(current_course)]
+    enrollment_status = get_enrollment(user.username, enrollment_course_id)
+    if not enrollment_status:
+        add_enrollment(user.username, enrollment_course_id)
+    elif not enrollment_status['is_active']:
+        update_enrollment(user.username, enrollment_course_id, is_active=True)
+
+def delete_state_of_review_problem(user, current_course, problem_id):
+    '''
+    Deletes the state of a review problem so it can be used infinitely
+    many times.
+
+    Parameters:
+        user (User): the current user interacting with the review xBlock
+        current_course (CourseLocator): The course the learner is currently in
+        problem_id (str?): The problem id whose state should be cleared
+    '''
+    pass
+
+def get_correctness_and_attempts(state):
+    '''
+    From the state of a problem from the Coursware Student Module,
+    determine if the learner correctly answered it initially and
+    the number of attempts they had for the original problem
+    
+    Parameter:
+        state (dict): The state of a problem
+
+    Returns a tuple of (correctness, attempts)
+        correctness (str): 'correct' or 'incorrect'
+        attempts (int): 0 if never attempted, else number of times attempted
+    Catches KeyError if learner never attempted the problem
+    '''
+    if state['score']['raw_earned'] == state['score']['raw_possible']:
+        correctness = 'correct'
+    else:
+        correctness = 'incorrect'
+
+    if 'attempts' in state:
+        attempts = state['attempts']
+    else:
+        attempts = 0
+
+    return (correctness, attempts)
+
+def is_correct_review_problem(user, current_course, problem_id):
+    '''
+    Checks a review problem to see if the learner has already correctly
+    answered it.
+
+    Returns True if the review problem was correctly answered, False otherwise
+    Catches IndexError if learner has never seen the review problem before.
+    Catches KeyError if learner has never attempted the review problem before.
```

Comment:
The above two lines would be better off as a comment in the `except` block below.